src/drawers/drawer.ts

```diff
@@ -76,7 +76,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.keyFunction()) {
```

Comment:
`Drawer`s are instantiated with a `dataset`.  It can be assumed that `this._dataset` always exists